rnd/autogpt_builder/src/components/CustomNode.tsx

```diff
@@ -161,26 +162,26 @@ const CustomNode: FC<NodeProps<CustomNodeData>> = ({ data, id }) => {
 
   // Helper function to parse keys with array indices
   const parseKeys = (key: string): ParsedKey[] => {
-    const regex = /(\w+)|\[(\d+)\]/g;
+    const splits = key.split(/_@_|_#_|_\$_|\./);
```

Comment:
instead of taking the key fragment using regex, we list all the separator and split them. Current serpator:
`.`, `_@_`, `_#_`, `_$_`